### PR TITLE
Custom3

### PR DESCRIPTION
**Pull request by tomyeh**
I moved the generation of *mailbox* from `_addressToString` to `Address.toString`, so users can override it with his own sanitization, if necessary.

### CODE REVIEW

File: lib/src/entities/message.dart [hunk 1]
```diff
@@ -71,6 +71,6 @@ class Message {
   String html;
   List<Attachment> attachments = [];
 
-  Iterable<Address> _asAddresses(Iterable<dynamic> as) =>
-      (as ?? []).map((a) => a is String ? Address(a) : a as Address);
+  static Iterable<Address> _asAddresses(Iterable<dynamic> adrs) =>
+      adrs == null ? []: adrs.map((a) => a is String ? Address(a) : a as Address);
```
Comment (close2) [inline lib/src/entities/message.dart:75]: Is this in some way better?
(I don't mind.  Honest question)

File: lib/src/entities/address.dart [hunk 2]
```diff
@@ -1,6 +1,13 @@
 class Address {
-  String name;
-  String mailAddress;
+  final String name;
+  final String mailAddress;
```
Comment (close2) [inline lib/src/entities/address.dart:3]: I can't add this change with a minor version bump.
Please revert this.  I'll try to keep this change in mind for a future version.

**Comment by close2** [comment]
I think that using `toString()` to build a text which must conform to the SMTProtocol isn't really correct.
Unfortunately I don't provide any sanitation hooks at the moment and will therefore accept the PR.

Please add a comment in the toString() function to make it clear that the output of the `toString` function must conform to the SMTP.

**Comment by close2** [comment]
Could you please also bump the version to 3.1.0 and adapt the Changelog.

File: lib/src/entities/message.dart (hunk 1 above)
Comment (tomyeh) [reply lib/src/entities/message.dart:75]: Technically, the performance shall be a little better since no need to pass `this` to the function.

**Comment by tomyeh** [comment]
To count on the implementation of `Address.toString()` to sanitize, `_validAddress()` can't validate `Address.name` directly. `toString()` might encode the name (such as wrapping with `"` or removing non-printable characters). Thus, adjust it a bit: https://github.com/kaisellgren/mailer/pull/139/commits/dcf7ba08be467ff4a76c5d84645018edf0ef676c

It doesn't look good. A proper hook mechanism is still a better solution. You can revoke it once you introduce a hook mechanism.

**Pull request merged**